sdk/metric/sdk.go

```diff
@@ -139,6 +143,15 @@ type (
 		labels        *label.Set
 		observed      export.Aggregator
 	}
+
+	// Implementations of MetricsProcessor can be provided as an config option to provide an opportunity
+	// to re-compose metrics labels based on the context when the metrics are recorded.
+	MetricsProcessor interface {
```

Comment:
`MetricsProcessor` does not seem like an appropriate name for this interface. It is overly general, uses the word "metrics" as a noun instead of an adjective (something that is avoided in OpenTelemetry), and duplicates the name of the other `metric.Processor` from the export package, which is is also a field of the `Accumulator`.

Can we use a more descriptive name of the function this interface will serve and possibly match broader [naming standards](https://golang.org/doc/effective_go.html#interface-names)?